### DISCUSSION TITLE
Bagaimana Cara Membuat “Recent Post” dengan Sumber Berdasarkan URL yang Difilter dari Penulis?

### DISCUSSION THREAD

**Discussion by yuliakbard** [mecha-cms/mecha#261] [Q&A] [answered]
Bagaimana cara membuat recent post dengan sumber nya berdasarkan url hasil dari extension author.

Rencananya ane mau menambahkan recent post di halaman user

``` php
<?php

$pages = Pages::from(LOT . D . 'page' . D . 'blog' . D . 'profile' . D . 'yuliakbard' . D . '1'); // benar gak yang ini kang?
$take = 6;
?>
<?php foreach($pages as $page):?>
<a href="<?=$page->url;?>"><?= $page->title;?></a>
    <?php endforeach;?>
```

trus metode lain yang mengambil data dari sumber dari feed json berdasarkan url yang di filtered by author, juga contohnya kalau boleh tau

**Comment by taufik-nurrohman** [comment, marked as answer]
Coba ini kalau di-debug keluar apa?

~~~ php
if ($site->is('user')) {
    $pages = Pages::from(LOT . D . 'page', 'page', true)->is(function ($v) use ($page) {
        $key = $v['author'] ?? "";
        if (!$key || '@' !== $key[0]) {
            return false;
        }
        return substr($key, 1) === $page->name;
    })->sort([-1, 'time'])->chunk(5, 0);
    test($pages->get()); // Ini keluar apa?
    exit;
}
~~~

**Reply by yuliakbard** [reply]
kalau di halaman user, tidak menampilkan apa-apa kang

**Reply by taufik-nurrohman** [reply]
Pastikan ada laman yang mengandung properti `author`:

~~~ yaml
---
title: asdf
author: '@yuliakbard'
...

asdf
~~~

**Reply by yuliakbard** [reply]
ada kang, properti `author` nya di setiap laman nya

![image](https://user-images.githubusercontent.com/96925342/222943821-0cd5d8ff-fdbd-40e3-84ee-f168feae4c36.png)

**Reply by yuliakbard** [reply]
wait, wait kang. seperti nya ane keliru penempatan kode, ane tempatin di `pages.php` harusnya di `page.php`

![image](https://user-images.githubusercontent.com/96925342/222944139-d7235f41-8ea2-48e5-b0fb-963920d90740.png)

**Reply by taufik-nurrohman** [reply]
Yaudah tinggal gini:

~~~ php
foreach ($pages as $p) {
    echo '<a href="' . $p->url . '">';
    echo $p->title;
    echo '</a>';
    echo '<br>';
}
~~~

**Reply by yuliakbard** [reply]
iya kang, sudah beres. makasih banyak kang
